index.js

```diff
@@ -174,7 +174,16 @@ InfluxDB.prototype.getDatabaseNames = function (callback) {
     if (err) {
       return callback(err, results)
     }
-    return callback(err, _.map(results[0].series[0].values, function (dbarray) { return dbarray[0] }))
+
+    if (_.isArray(results) && !_.isEmpty(results) && _.isArray(results[0].series) &&
+        !_.isEmpty(results[0].series) && _.isArray(results[0].series[0].values)) {
+
+      return callback(err, _.map(results[0].series[0].values, function (dbarray) {
+        if (_.isArray(dbarray) && !_.isEmpty(dbarray)) {
+          return dbarray[0]
+        }
+      }))
+    } else { return callback(new Error('bad response from server', results))}
```

Comment:
My intention wasn't to guard from an attacker as if an attacker can intercept a response your well past screwed at that stage. I meant crafted as in my example ones to show the point of should this lib blow up without being able to be caught by the caller.

I haven't read the influx server code but It was to guard from Influx sending bad data (as I'm in not sure if Influx server can send the second response back with an undefined/null as that seemed plausible to me).